libs/libapi/src/libapi/duckdb.py

```diff
@@ -67,25 +72,39 @@ def get_download_folder(root_directory: StrPath, dataset: str, revision: str, co
     return f"{root_directory}/{DUCKDB_INDEX_DOWNLOADS_SUBDIRECTORY}/{DUCKDB_VERSION}/{subdirectory}"
 
 
+def available_disk_space(save_path: str, required_space: int) -> bool:
+    disk_stat = os.statvfs(save_path)
+    # Calculate free space in bytes
+    free_space = disk_stat.f_bavail * disk_stat.f_frsize
+    return free_space >= required_space
+
+
 def download_index_file(
     cache_folder: str,
     index_folder: str,
     target_revision: str,
+    size_bytes: int,
     dataset: str,
     repo_file_location: str,
     hf_token: Optional[str] = None,
 ) -> None:
     logging.info(f"init_dir {index_folder}")
-    init_dir(index_folder)
-    download_file_from_hub(
-        repo_type=REPO_TYPE,
-        revision=target_revision,
-        repo_id=dataset,
-        filename=repo_file_location,
-        local_dir=index_folder,
-        hf_token=hf_token,
-        cache_dir=cache_folder,
-    )
+    if not available_disk_space(cache_folder, size_bytes):
+        raise DownloadIndexError("Unable to download index file")  # Should we show the remaining disk space?
+    try:
+        init_dir(index_folder)
+        download_file_from_hub(
+            repo_type=REPO_TYPE,
+            revision=target_revision,
+            repo_id=dataset,
+            filename=repo_file_location,
+            local_dir=index_folder,
+            hf_token=hf_token,
+            cache_dir=cache_folder,
+        )
+    except OSError as err:
+        if err.errno == errno.ENOSPC:
+            raise DownloadIndexError("Unable to download index file", err)
```

Comment:
Same here. Even if the process had enough space before downloading, other workers could download files, which might lead to a lack of space. Again, I am unsure about what to say to the user.